src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -1244,8 +1245,36 @@ export const createTwilioTransaction = async (
   body: CreateSecretRequest,
   session: ClientSession,
 ): Promise<void> => {
+  const logMeta = {
+    action: 'updateTwilioCredentials',
+    msgSrvcName,
+  }
   await form.updateMsgSrvcName(msgSrvcName, session)
-  await secretsManager.createSecret(body).promise()
+  try {
+    await secretsManager.createSecret(body).promise()
+  } catch (err) {
+    const awsError = err as AWSError
+
+    if (awsError.code === 'ResourceExistsException') {
+      logger.info({
+        message:
+          'Twilio Credentials exists in Secrets Manager, proceeeding to restore and update secret',
+        meta: logMeta,
+      })
+
+      const restoreRequestBody: RestoreSecretRequest = {
+        SecretId: msgSrvcName,
+      }
+
+      const putRequestBody: PutSecretValueRequest = {
+        SecretId: msgSrvcName,
+        SecretString: body.SecretString,
+      }
+
+      await secretsManager.restoreSecret(restoreRequestBody).promise()
+      await secretsManager.putSecretValue(putRequestBody).promise()
+    }
+  }
```

Comment:
should we convert this whole block to a neverthrow chain? seems like the catch has a possibility of throwing again, resulting in a need for nested try catches.

Reply:
Just wanted to check why is a neverthrow chain a better option here, cause I am thinking that the neverthrow chain might have some nesting of ResultAsync.fromPromise too (correct me if I'm wrong 😅 )

Reply:
because this means we can reduce the possibility of missing a try-catch block and have errors bubbling and crashing the application. I also don't think the nesting is that bad; 

```ts
return (
  ResultAsync.fromPromise(
    secretsManager.createSecret(body).promise(),
    (error: AWSError) => error
  ).orElse((error) => {
    if (error.code === 'ResourceExistsException') {
      logger.info(...)
      return ResultAsync.fromPromise(
        secretsManager.restoreSecret(restoreRequestBody).promise(),
        error => error
      )
      // Restore success, continue to update request body
      .andThen(() => 
        ResultAsync.fromPromise(
        secretsManager.putSecretValue(putRequestBody).promise(),
        error => error
      )
    }
    // Unknown error, return directly
    return errAsync(new SomeAwsError(error))
  })
)
```

Reply:
I tried and tested this implementation but realised that there is an issue which is that the transaction will not be rollbacked if an error occurs with AWS Secrets Manager. This is because the error is wrapped around `ResultAsync<>` and as such, will never be "thrown" from within the `createTwilionTransaction` helper function.

Do you have any idea how we can extract the error or should we still continue with this method?